### PR TITLE
Implement Odds Command

### PR DESCRIPTION
**Pull request by UnlimitedFrames**

### CODE REVIEW

**Review by mrvillage — DISMISSED** [review]
Great work! Just had a couple small comments/future notes. :)

File: bot/src/commands/odds.rs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+use crate::types::{Context, Error};
+
+#[poise::command(slash_command, prefix_command)]
+async fn odds(
+    ctx: Context<'_>,
+    #[description = "Nation Initiating the attack"] attacking_nation: String,
+    #[description = "Nation Defending the attack"] defending_nation: String,
+) -> Result<(), Error> {
+    let attr_res = Nation::convert(&ctx, attacking_nation).await;
+    let def_res = Nation::convert(&ctx, defending_nation).await;
```
[the remark this answers is not in the document]
Comment (UnlimitedFrames) [reply bot/src/commands/odds.rs:14]: shhhhh no commenty yet

File: bot/src/commands/odds.rs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+            if att * att_mod > def * def_mod {
+                wins += 1.0
+            }
+            total += 1.0;
+        }
+    }
+
+    return wins / total;
+}
```
[the remark this answers is not in the document]
Comment (UnlimitedFrames) [reply bot/src/commands/odds.rs:77]: there definitally is!!! but shhhhhh give me a min u werent meant to see this

[the remark this answers is not in the document]
Comment (UnlimitedFrames) [reply bot/src/commands/odds.rs:77]: also there kinda is? but its a much more complicated problem than it looks, and doing it like this is easy and fast

File: bot/src/embeds/pnw.rs [hunk 3]
```diff
@@ -266,3 +266,52 @@ pub fn nation<'a>(
...
+            ])
+            .description(format!(
+                "{} vs {}",
+                strings::link(
+                    attr.name.clone(),
+                    format!("{}{}", "https://politicsandwar.com/nation/id=", attr.id),
+                ),
+                strings::link(
+                    def.name.clone(),
+                    format!("{}{ }", "https://politicsandwar.com/nation/id=", def.id),
```
Comment (mrvillage) [inline bot/src/embeds/pnw.rs:299]:
```suggestion
                    format!("https://politicsandwar.com/nation/id={}", def.id),
```

File: bot/src/embeds/pnw.rs [hunk 4]
```diff
@@ -266,3 +266,52 @@ pub fn nation<'a>(
...
+) -> impl Fn(&mut CreateEmbed) -> &mut CreateEmbed + 'a {
+    move |e: &mut CreateEmbed| {
+        e.colour(consts::embed::INFO_EMBED_COLOUR)
+            .fields([
+                ("Ground Battle", format_chance(ground_chance), true),
+                ("Air Strike", format_chance(air_chance), true),
+                ("Naval Battle", format_chance(naval_chance), true),
+                ("Nuke", percent(nuke_chances), true),
+                ("Missile", percent(missile_chances), true),
+                ("Spy Op", "Not Implimented".to_string(), true),
```
Comment (mrvillage) [inline bot/src/embeds/pnw.rs:289]:
```suggestion
                ("Spy Op", "Not Implemented".to_string(), true),
```

**Review by mrvillage — CHANGES_REQUESTED** [review]

**Review by mrvillage — APPROVED** [review]

**Pull request merged**